File: java/Visitor/src/Customer.java

import java.util.ArrayList;

public class Customer implements Visitable {

	String name;

	private ArrayList<Order> orders = new ArrayList<Order>();

	public void accept(Visitor visitor) {
		visitor.visit(this);	
		for (Order eachOrder: orders){
			eachOrder.accept(visitor);
		}
	}

	public void addOrder(Order order) {
		orders.add(order);
	}

	public Customer(String name) {
		this.name = name;
	}
}

File: ruby/Visitor/customer.rb
require_relative 'visitable'
require_relative 'visitor'
require_relative 'order'

class Customer < Visitable
  attr_accessor :name
  def initialize(name)
    @name = name
    @orders = Array.new
  end

  def accept(visitor)
    visitor.visit(self)
    for order in @orders
      order.accept(visitor)
    end
  end

  def add_order(order)
    @orders.push(order)
  end
end